compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -15784,8 +15799,9 @@ OMR::Z::TreeEvaluator::arraycmpSIMDHelper(TR::Node *node,
       if(isArrayCmpLen)
          {
          // Return 0-based index of first non-matching element
-         // resultReg - lastByteIndexReg = number of elements compared before the last loop
+         // (resultReg - 1) - lastByteIndexReg = number of elements compared before the last loop
          // vectorOutputReg contains the 0-based index of the first non-matching element (index is its position in the vector)
+         generateRILInstruction(cg, TR::InstOpCode::getSubtractLogicalImmOpCode(), node, resultReg, 1);
          generateRRInstruction(cg, TR::InstOpCode::getSubstractRegOpCode(), node, resultReg, lastByteIndexReg);
          generateVRScInstruction(cg, TR::InstOpCode::VLGV/*B*/, node, lastByteIndexReg, vectorOutputReg, generateS390MemoryReference(7, cg), 0);
          generateRRInstruction(cg, TR::InstOpCode::getAddRegOpCode(), node, resultReg, lastByteIndexReg);
```

Comment:
If LA could support negative offset, I could use load address to result reg with lastByteIndexReg as base and -1 as offset instead of line 15804 and this

Reply:
Hmm, LAY do support signed displacement. But even in that case, it follows address arithmetic. So you may run into issues on 32-bit platforms. 

Though, one general improvement I can suggest here that we can improve on existing code is to use `AHI` and `AGHI`. Operands are treated signed, so you should be able to subtract -1 easily. Compared to SubtractLogicalImmOp, it uses 4 bytes only.

Reply:
I changed all subtract instructions to add immediate. Its a separate commit for now until I test and you check the change.